rocksdb/src/move.rs

```diff
@@ -0,0 +1,100 @@
+use {
+    bytes::Bytes,
+    eth_trie::DB,
+    move_binary_format::{errors::PartialVMError, CompiledModule},
+    move_core_types::{
+        account_address::AccountAddress,
+        language_storage::{ModuleId, StructTag},
+        metadata::Metadata,
+        resolver::{resource_size, ModuleResolver, ResourceResolver},
+        value::MoveTypeLayout,
+    },
+    move_table_extension::{TableHandle, TableResolver},
+    rocksdb::DB as RocksDb,
+};
+
+pub struct RocksEthTrieDb<'db> {
+    db: &'db RocksDb,
+}
+
+impl<'db> RocksEthTrieDb<'db> {
+    pub fn new(db: &'db RocksDb) -> Self {
+        Self { db }
+    }
+
+    pub fn db(&self) -> &'db RocksDb {
+        self.db
+    }
+}
+
+impl<'db> ModuleResolver for RocksEthTrieDb<'db> {
+    type Error = rocksdb::Error;
+
+    fn get_module_metadata(&self, module_id: &ModuleId) -> Vec<Metadata> {
+        if let Ok(Some(bytes)) = self.get_module(module_id) {
+            if let Ok(module) = CompiledModule::deserialize(&bytes) {
+                return module.metadata;
+            }
+        }
+
+        Vec::new()
+    }
+
+    fn get_module(&self, id: &ModuleId) -> Result<Option<Bytes>, Self::Error> {
+        let key = id.address().to_canonical_string();
```

Comment:
As discussed on the call, we'll need to do something to distinguish different kinds of keys.

For this particular case the key you are using is not enough because a single address can have multiple different modules. This is why `ModuleId` also includes a `name` field.

But all of this is secondary to the point that I don't think the resolver traits are needed for the `RocksDB` type because we will only be accessing the storage via the trie. So only the `HistoricResolver` needs to implement the resolver traits.